### PR TITLE
Add Benchmark command line reference

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _benchmark/commands/execute-test.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+- `--test-execution-id`: Defines a unique ID for this test run.
+- `--test-procedure`: Defines a test procedure to use. You can find a list of test procedures using `opensearch-benchmark list test-procedures`.
+- `--include-tasks`: Defines a comma-separated list of test procedure tasks to run. By default, all tasks listed in a test procedure array are run.
+- `--exclude-tasks`: Defines a comma-separated list of test procedure tasks not to run.
+- `--enable-assertions`: Enables assertion checks for tasks. Default is `false`.
+
+### Pipelines
+
+The `--pipeline` option selects a pipeline to run.
```
Comment (rishabh6788) [inline _benchmark/commands/execute-test.md:153]: Is there a command to list the type of pipelines as well?
If yes, can we add that command here?

File: _benchmark/commands/execute-test.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
+---
+layout: default
+title: execute-test
+nav_order: 65
+parent: Command reference
+---
+
+# execute_test
```
Comment (IanHoang) [inline _benchmark/commands/execute-test.md:8]: "execute_test" --> "execute-test" as of 0.5.0

File: _benchmark/commands/execute-test.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+The following options set which version OpenSearch and OpenSearch plugins the benchmark test uses.
+
+- `--distribution-version`: Downloads the specified OpenSearch distribution based on version number. For a list of released OpenSearch versions, see [Version history](https://opensearch.org/docs/version-history/).
+- `--distribution-repository`: Defines the repository from where the OpenSearch distribution should be downloaded. Default is `release`.
+- `--revision`: Defines the current source code revision for which to run a benchmark test. Default is `current`.
+   - `current`: Uses the source tree's current revision based on your OpenSearch distribution. 
+   - `latest`: Fetches the latest revision from the main branch of the source tree.
+   - You can also a time stamp or commit ID from the source tree. When using a timestamp, specify `@ts` where "ts" is a valid ISO 8601 timestamp, for example `@2013-07-27T10:37:00Z`.
+-  `--opensearch-plugins`: Defines which [OpenSearch plugins]({{site.url}}{{site.baseurl}}/install-and-configure/plugins/) to install. By default, not plugins are installed.
```
Comment (IanHoang) [inline _benchmark/commands/execute-test.md:108]: "not plugins" --> "no plugins"

File: _benchmark/commands/execute-test.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+- `--plugin-params:` Defines a comma-separated list of key:value pairs that are injected verbatim into all plugins as variables.
+- `--runtime-jdk`: The major version of JDK to use.
+- `--client-options`: Defines a comma-separated lists of clients to use. All options are passed to the OpenSearch Python client. Default is `timeout:60`.
+
+### Cluster
+
+The following options set which cluster options.
```
Comment (IanHoang) [inline _benchmark/commands/execute-test.md:115]: Super nit: "The following options et cluster options"

File: _benchmark/commands/execute-test.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+The following options determine which workload is used to run the test.
+
+- `--workload-repository`: Defines the repository from where OpenSearch Benchmark loads workloads.
+- `--workload-path`: Defines the path to a downloaded or custom workload.
+- `--workload-revision`: Defines a specific revision from the workload source tree that OpenSearch Benchmark should use.
+- `--workload`: Defines the workload to use based on the workload's name. You can find a list of preloaded workloads using `opensearch-benchmark list workloads`.
+
+### Test procedures
+
+The following options define what test procedures the test uses and which tasks are contained inside the procedure.
```
Comment (IanHoang) [inline _benchmark/commands/execute-test.md:149]: Could switch `tasks` to `operations`?

File: _benchmark/commands/execute-test.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+### Errors
+
+The following options set how OpenSearch Benchmark handles errors when running tests.
+
+- `--on-error`: Controls how OpenSearch Benchmark responds to errors. Default is `continue`.
+  - `continue`: Continues to run the test despite the error.
+  - `abort`: Aborts the test when an error occurs.
+- `--preserve-install`: Keeps the Benchmark candidate and its index. Default is `false`.
+- `--kill-running-processes`: When set to `true`, stops any OpenSearch processes currently running and allows OpenSearch Benchmark to continue to run. Default is `false`.
```
Comment (IanHoang) [inline _benchmark/commands/execute-test.md:178]: Should be "stops any OpenSearch Benchmark processes"

File: _benchmark/commands/list.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+- `workloads`: Workloads
+- `pipelines`: Pipelines
+- `test_executions`: Single run of a workload
+- `provision_config_instances`: Provisioned configuration instances
+- `opensearch-plugins`: OpenSearch plugins
+
+
+## Usage
+
+The following example lists tests ran inside a workload:
```
Comment (IanHoang) [inline _benchmark/commands/list.md:22]: Super nit: "ran with a workload"

**Review by IanHoang — APPROVED** [review]
Added a few comments but overall, LGTM! Thanks Nate.

File: _benchmark/commands/execute-test.md [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+|  100th percentile service time |     painless_dynamic |   459.474 |     ms |
+
+----------------------------------
+[INFO] SUCCESS (took 2634 seconds)
+----------------------------------
+```
+
+## Options
+
+Use the following options to customize the `execute-test` command for your use case. Options in this section are categorized by their use-case.
```
Comment (cwillum) [inline _benchmark/commands/execute-test.md:84]: use case. no hyphen necessary.

File: _benchmark/commands/download.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+- `--provision-config-repository`: Defines the repository from which OpenSearch Benchmark loads `provision-configs` and `provision-config-instances`. 
+- `--provision-config-revision`: Defines a specific Git revision in the `provision-config` that OpenSearch Benchmark should use.
+- `--provision-config-path`: Defines the path to the `--provision-config-instance` and any OpenSearch plugin configurations to use.
+- `--distribution-version`: Downloads the specified OpenSearch distribution based on version number. For a list of released OpenSearch versions, see [Version history](https://opensearch.org/docs/version-history/).
+- `--distribution-repository`: Defines the repository from where the OpenSearch distribution should be downloaded. Default is `release`.
+- `--provision-config-instance`: Defines the `--provision-config-instance` to use. You can see possible configuration instances using the command `opensearch-benchmark list provision-config-instances`. 
+- `--provision-config-instance-params`: A comma-separated list of key:value pairs injected verbatim as variables for the `provision-config-instance`.
```
Comment (cwillum) [inline _benchmark/commands/download.md:38]: key-value pairs, as in line 91?

File: _benchmark/commands/execute-test.md [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+
+## General settings
+
+The following options shape how each test runs and how results are output.
+
+- `--test-mode`: Runs the given workload in test mode, which is useful when checking a workload for errors.
+- `--user-tag`: Defines user-specific key-value pairs to be used in metric record as meta information, for example, `intention:baseline-ticket-12345`.
```
Comment (cwillum) [inline _benchmark/commands/execute-test.md:91]: yes, it looks like "key-value pair" is the preferred formatting across the web. (with hyphen)

File: _benchmark/commands/execute-test.md [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+- `--results-format`: Defines the output format for the command line results, either `markdown` or `csv`. Default is `markdown`.
+- `--results-number-align`: Defines the column number alignment for when the `compare` command outputs results. Default is `right`.
+- `--results-file`: When provided a file path, writes the compare results to the file indicated.
```
Comment (cwillum) [inline _benchmark/commands/execute-test.md:94]:
```suggestion
- `--results-file`: When provided a file path, writes the compare results to the file indicated in the path.
```

File: _benchmark/commands/execute-test.md [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+- `--show-in-results`: Determines whether or not to include the comparison in the results file.
+
+
+### Distributions
+
+The following options set which version OpenSearch and OpenSearch plugins the benchmark test uses.
+
+- `--distribution-version`: Downloads the specified OpenSearch distribution based on version number. For a list of released OpenSearch versions, see [Version history](https://opensearch.org/docs/version-history/).
+- `--distribution-repository`: Defines the repository from where the OpenSearch distribution should be downloaded. Default is `release`.
+- `--revision`: Defines the current source code revision for which to run a benchmark test. Default is `current`.
```
Comment (cwillum) [inline _benchmark/commands/execute-test.md:104]:
```suggestion
- `--revision`: Defines the current source code revision to use for running a benchmark test. Default is `current`.
```

File: _benchmark/commands/execute-test.md [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+   - `current`: Uses the source tree's current revision based on your OpenSearch distribution. 
+   - `latest`: Fetches the latest revision from the main branch of the source tree.
+   - You can also a time stamp or commit ID from the source tree. When using a timestamp, specify `@ts` where "ts" is a valid ISO 8601 timestamp, for example `@2013-07-27T10:37:00Z`.
```
Comment (cwillum) [inline _benchmark/commands/execute-test.md:107]:
```suggestion
   - You can also use a time stamp or commit ID from the source tree. When using a timestamp, specify `@ts` where "ts" is a valid ISO 8601 timestamp, for example `@2013-07-27T10:37:00Z`.
```
Does this description belong with line 106?

File: _benchmark/commands/execute-test.md [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+-  `--opensearch-plugins`: Defines which [OpenSearch plugins]({{site.url}}{{site.baseurl}}/install-and-configure/plugins/) to install. By default, no plugins are installed.
+- `--plugin-params:` Defines a comma-separated list of key:value pairs that are injected verbatim into all plugins as variables.
```
Comment (cwillum) [inline _benchmark/commands/execute-test.md:109]: key-value pairs

File: _benchmark/commands/execute-test.md [hunk 15]
```diff
@@ -0,0 +1,178 @@
...
+- `--runtime-jdk`: The major version of JDK to use.
+- `--client-options`: Defines a comma-separated lists of clients to use. All options are passed to the OpenSearch Python client. Default is `timeout:60`.
```
Comment (cwillum) [inline _benchmark/commands/execute-test.md:111]:
```suggestion
- `--client-options`: Defines a comma-separated list of clients to use. All options are passed to the OpenSearch Python client. Default is `timeout:60`.
```

File: _benchmark/commands/execute-test.md [hunk 16]
```diff
@@ -0,0 +1,178 @@
...
+
+### Cluster
+
+The following options relate to the target cluster of the benchmark.
```
Comment (cwillum) [inline _benchmark/commands/execute-test.md:115]: Is the intention that more options will be added at some time in the future? You may still want to be accurate and say "following options relates to ..."

File: _benchmark/commands/execute-test.md [hunk 17]
```diff
@@ -0,0 +1,178 @@
...
+
+- `--target-hosts`: Defines a comma-separated list of host:port pairs which should be targeted if using the pipeline `benchmark-only`. Default is `localhost:9200`.
```
Comment (cwillum) [inline _benchmark/commands/execute-test.md:117]: host-port pairs?

File: _benchmark/commands/execute-test.md [hunk 18]
```diff
@@ -0,0 +1,178 @@
...
+
+
+### Distributed workload generation
+
+The following options help those who want to use multiple hosts to generate load to the benchmark cluster.
+
+- `--load-worker-coordinator-hosts`: Defines a comma-separated list of hosts that coordinate loads. Default is `localhost`.
+- `--enable-worker-coordinator-profiling`: Enables an analysis of the performance of call in OpenSearch Benchmark's worker coordinator. Default is `false`.
```
Comment (cwillum) [inline _benchmark/commands/execute-test.md:125]:
```suggestion
- `--enable-worker-coordinator-profiling`: Enables an analysis of the performance of a call in OpenSearch Benchmark's worker coordinator. Default is `false`.
```

File: _benchmark/commands/execute-test.md [hunk 19]
```diff
@@ -0,0 +1,178 @@
...
+
+### Provisioning
+
+The following options help customize how OpenSearch Benchmark provisions OpenSearch and workloads.
+
+- `--provision-config-repository`: Defines the repository from which OpenSearch Benchmark loads `provision-configs` and `provision-config-instances`. 
+- `--provision-config-path`: Defines the path to the `--provision-config-instance` and any OpenSearch plugin configurations to use.
+- `--provision-config-revision`: Defines a specific Git revision in the `provision-config` that OpenSearch Benchmark should use.
+- `--provision-config-instance`: Defines the `--provision-config-instance` to use. You can see possible configuration instances using the command `opensearch-benchmark list provision-config-instances`. 
+- `--provision-config-instance-params`: A comma-separated list of key:value pairs injected verbatim as variables for the `provision-config-instance`.
```
Comment (cwillum) [inline _benchmark/commands/execute-test.md:135]: key-value pairs

File: _benchmark/commands/execute-test.md [hunk 20]
```diff
@@ -0,0 +1,178 @@
...
+
+The `--pipeline` option selects a pipeline to run. You can find a list of pipelines supported by OpenSearch Benchmark by running `opensearch-benchmark list pipelines`.
+
+
+### Telemetry
+
+The following options enable telemetry devices on OpenSearch Benchmark.
+
+- `--telemetry`: Enables the provided telemetry devices when the devices are provided using a comma-separated list. You can find a list of possible telemetry devices by using `opensearch-benchmark list telemetry`.
+- `--telemetry-params`: Defines a comma-separated list of key:value pairs that are injected verbatim into the telemetry devices as parameters.
```
Comment (cwillum) [inline _benchmark/commands/execute-test.md:167]: key-value

File: _benchmark/commands/index.md [hunk 21]
```diff
@@ -0,0 +1,26 @@
+---
+layout: default
+title: Command reference
+nav_order: 50
+has_children: true
+---
+
+# OpenSearch Benchmark command reference
+
+This section provides a list of command supported by OpenSearch Benchmark, including for commonly used commands such as `execute-test` and `list`.
```
Comment (cwillum) [inline _benchmark/commands/index.md:10]:
```suggestion
This section provides a list of commands supported by OpenSearch Benchmark, including commonly used commands such as `execute-test` and `list`.
```

File: _benchmark/commands/info.md [hunk 22]
```diff
@@ -0,0 +1,158 @@
...
+
+
+-------------------------------
+[INFO] SUCCESS (took 2 seconds)
+-------------------------------
+```
+
+## Options
+
+You can use the following options with the `info` command.
```
Comment (cwillum) [inline _benchmark/commands/info.md:149]:
```suggestion
You can use the following options with the `info` command:
```
Extra space after this?

File: _benchmark/commands/list.md [hunk 23]
```diff
@@ -0,0 +1,70 @@
...
+78602e07-0ff8-4f00-9a0e-746fb64e4129  20230519T193258Z           geonames                           append-no-conflicts  external                                30260cf
+
+-------------------------------
+[INFO] SUCCESS (took 0 seconds)
+-------------------------------
+```
+
+## Options
+
+You can use the following options with the `test` command.
```
Comment (cwillum) [inline _benchmark/commands/list.md:63]:
```suggestion
You can use the following options with the `test` command:
```

**Review by cwillum — APPROVED** [review]
Looks good.

File: _benchmark/commands/execute-test.md (hunk 8 above)
Comment (Naarcha-AWS) [inline _benchmark/commands/execute-test.md:84]:
```suggestion
Use the following options to customize the `execute-test` command for your use case. Options in this section are categorized by their use case.
```

File: _benchmark/commands/download.md (hunk 9 above)
Comment (Naarcha-AWS) [inline _benchmark/commands/download.md:38]:
```suggestion
- `--provision-config-instance-params`: A comma-separated list of key-value pairs injected verbatim as variables for the `provision-config-instance`.
```

File: _benchmark/commands/execute-test.md (hunk 16 above)
Comment (Naarcha-AWS) [inline _benchmark/commands/execute-test.md:115]:
```suggestion
The following option relates to the target cluster of the benchmark.
```

File: _benchmark/commands/execute-test.md (hunk 17 above)
Comment (Naarcha-AWS) [inline _benchmark/commands/execute-test.md:117]:
```suggestion
- `--target-hosts`: Defines a comma-separated list of host-port pairs which should be targeted if using the pipeline `benchmark-only`. Default is `localhost:9200`.
```

File: _benchmark/commands/execute-test.md (hunk 18 above)
Comment (Naarcha-AWS) [inline _benchmark/commands/execute-test.md:125]:
```suggestion
- `--enable-worker-coordinator-profiling`: Enables an analysis of the performance of OpenSearch Benchmark's worker coordinator. Default is `false`.
```

File: _benchmark/commands/execute-test.md (hunk 19 above)
Comment (Naarcha-AWS) [inline _benchmark/commands/execute-test.md:135]:
```suggestion
- `--provision-config-instance-params`: A comma-separated list of key-value pairs injected verbatim as variables for the `provision-config-instance`.
```

File: _benchmark/commands/execute-test.md (hunk 20 above)
Comment (Naarcha-AWS) [inline _benchmark/commands/execute-test.md:167]:
```suggestion
- `--telemetry-params`: Defines a comma-separated list of key-value pairs that are injected verbatim into the telemetry devices as parameters.
```

File: _benchmark/commands/compare.md [hunk 24]
```diff
@@ -0,0 +1,132 @@
+---
+layout: default
+title: compare
+nav_order: 55
+parent: Command reference
+---
+
+# compare
```
[the remark this answers is not in the document]
Comment (Naarcha-AWS) [reply _benchmark/commands/compare.md:8]: Dev's preference is to have this match how it appears on the command-line. We had a similar precedent with Data Prepper.

**Pull request merged**